File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String name : participant) {
            if (map.get(name) != null) {
                int num = map.get(name);
                num++;
                map.put(name, num);
            } else {
             map.put(name, 1);   
            }
        }
        
        for (String name : completion) {
            int num = map.get(name);
            num--;
            map.put(name, num);
        }
        
        for (String name : participant) {
            if (map.get(name) != 0) {
                answer = name;
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    answer = ''
    
    p = Counter(participant)
    
    for c in completion:
        p[c] -= 1
        if p[c] == 0:
            del p[c]
    
    answer = list(p.keys())[0]
    
    return answer